crates/jstz_proto/src/executor/smart_function.rs

```diff
@@ -470,27 +475,43 @@ impl HostScript {
         Ok(amount)
     }
 
-    /// Transfer xtz from `src` to `dst` if the `TRANSFER_HEADER_KEY` header is present
-    /// and the transfer amount > 0 in request
+    fn verify_headers(headers: &impl Deref<Target = Headers>) -> JsResult<()> {
+        if headers.contains_key(X_JSTZ_AMOUNT) {
+            return Err(JsError::from_native(
+                JsNativeError::typ().with_message("Invalid header X-JSTZ-AMOUNT"),
```

Comment:
```suggestion
                JsNativeError::typ().with_message("X-JSTZ-AMOUNT header should not be present"),
```

Reply:
```suggestion
                JsNativeError::error().with_message("Invalid header X-JSTZ-AMOUNT"),
```

I don't think this should be a type error and there's nothing like `ValueError` in JavaScript, so a general error is probably better.